test/kafka/kafka_setup.sh

```diff
@@ -80,27 +80,36 @@ function create_tls_secrets() {
     --from-literal=ca.crt="$STRIMZI_CRT" \
     --from-literal=user.crt="$TLSUSER_CRT" \
     --from-literal=user.key="$TLSUSER_KEY" \
+    --from-literal=protocol="SSL" \
     --dry-run=client -o yaml | kubectl apply -n "${SYSTEM_NAMESPACE}" -f -
 }
 
 function create_sasl_secrets() {
   ca_cert_secret="my-cluster-cluster-ca-cert"
   sasl_user="my-sasl-user"
 
-  echo "Waiting until secrets: ${ca_cert_secret}, ${sasl_user} don't exist"
+  echo "Waiting until secrets: ${ca_cert_secret}, ${sasl_user} exist"
   wait_until_object_exists secret ${ca_cert_secret} kafka
   wait_until_object_exists secret ${sasl_user} kafka
 
-  echo "Creating SASL Kafka secret"
+  echo "Creating SASL_SSL and SASL_PLAINTEXT Kafka secrets"
 
   STRIMZI_CRT=$(kubectl -n kafka get secret ${ca_cert_secret} --template='{{index .data "ca.crt"}}' | base64 --decode )
   SASL_PASSWD=$(kubectl -n kafka get secret ${sasl_user} --template='{{index .data "password"}}' | base64 --decode )
 
   kubectl create secret --namespace "${SYSTEM_NAMESPACE}" generic strimzi-sasl-secret \
     --from-literal=ca.crt="$STRIMZI_CRT" \
     --from-literal=password="$SASL_PASSWD" \
-    --from-literal=saslType="SCRAM-SHA-512" \
```

Comment:
I think this would break KafkaSource tests.

Reply:
Hm, interesting...

I added these below:
```
--from-literal=protocol="SASL_SSL" \
--from-literal=sasl.mechanism="SCRAM-SHA-512" \
```

I thought these are necessary and `saslType` is actually not used.
Can't really find where `saslType` is read from the secret.

However, then on the other hand; how was it working before?

Reply:
@pierDipi Yes, that change breaks KafkaSource. But, I don't see how.

Reply:
It's hard-coded here in `eventing-kafka`,
https://github.com/knative-sandbox/eventing-kafka/blob/15a64fb7261fdb9e66bb2f7ce8171a49b6df4641/test/e2e/helpers/kafka_source.go#L326